font/src/ft/fc/pattern.rs

```diff
@@ -18,16 +18,15 @@ use std::path::PathBuf;
 use std::ptr::{self, NonNull};
 use std::str;
 
-use foreign_types::{ForeignType, ForeignTypeRef};
+use fontconfig::fontconfig::{
+    FcBool, FcChar8, FcConfigSubstitute, FcDefaultSubstitute, FcFontRenderPrepare, FcPattern,
+    FcPatternAddCharSet, FcPatternAddDouble, FcPatternAddInteger, FcPatternAddString,
+    FcPatternCreate, FcPatternDestroy, FcPatternGetBool, FcPatternGetDouble, FcPatternGetInteger,
+    FcPatternGetString, FcPatternPrint, FcResultMatch,
+};
```

Comment:
Here we use `fontconfig::fontconfig` again, instead of `super::ffi`, which is inconsistent.

Generally I think that these `fontconfig::fontconfig` imports are an insane mess. We have cross-module depedencies for the sole reason of using a crate import rename which seems to originate from the old edition, but just doesn't work well with the new 2018 edition.

I think the reasonable thing to do would be to import `fontconfig::fontconfig` everywhere. There's just no way around that and it's just imports so it shouldn't matter too much. If we ever import `fontconfig::fontconfig::self` anywhere, then we can rename that to `fontconfig::fontconfig::self as fontconfig_ffi`.

However we should never use `super::fontconfig::fontconfig` or `self::fontconfig::fontconfig` anywhere, since that just unreasonably complicates the cross-module depedencies for no reason.

Sorry for going back and forth on this, but if we want to have a consistent code style, we need to make sure it's intuitive and `super::ffi::` definitely is not that.